Non root install

mr-c opened the pull request:
PR so @mr-c can follow along

themangoemoji committed:
Added non root install instructions

themangoemoji committed:
EC2 non root install instructions

themangoemoji committed:
pipeline notes and prokka install update

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -58,12 +58,17 @@ OK, now we have installed almost all of the software we need, hurrah!
 First, check out the source repository and grab the (...large) initial data
 sets::
 
+ echo "export PATH=${PATH}:~/bin" >> ~/.bashrc
```
and run it right here as well
`export PATH=${PATH}:~/bin`

mr-c commented on pipeline/install-prokka.sh:
```diff
@@ -6,41 +6,41 @@ cd /mnt
 cd aragorn1.2.36/
 gcc -O3 -ffast-math -finline-functions -o aragorn aragorn1.2.36.c
-cp aragorn /usr/local/bin
+cp aragorn ~/bin
```
When copying to a directory I usually use a trailing slash for clarity

themangoemoji committed:
pipline note update

themangoemoji committed:
changed ~ to /Users/narrows

themangoemoji committed:
rename

themangoemoji committed:
no mo .profile

themangoemoji committed:
no mo .profile

themangoemoji committed:
Merge branch 'non-root-install' of github.com:ged-lab/2012-paper-diginorm into non-root-install

themangoemoji committed:
Add RackSpace install instructions

*Add virtualenv for non-root easyinstall

themangoemoji committed:
Add RackSpace install instructions

*Add virtualenv for non-root easyinstall

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -22,37 +22,56 @@ Make sure you edit your security groups to include port 22 (SSH) and port
-
- sudo su
+If you are running on RackSpace, we will need to add extra space for data by formating and mounting a drive.
+Formating/Mounting Drive::
```
You're using a '15 GB I/O v1' flavor Rackspace cloud server

themangoemoji commented on pipeline/install-prokka.sh:
```diff
@@ -6,41 +6,41 @@ cd /mnt
 curl -O ftp://ftp.ncbi.nih.gov/toolbox/ncbi_tools/converters/by_program/tbl2asn/linux64.tbl2asn.gz
 gunzip linux64.tbl2asn.gz
 mv linux64.tbl2asn tbl2asn
 chmod +x tbl2asn
```
@mr-c this and several other actions in this file require root.
Maybe I can extract these to the beginning?

mr-c replied:
Which lines require root? I don't see any.

themangoemoji replied:
The chmod, no? Its halting on this install anyways
in particular 
`prokka-1.7/db/hmm: Cannot utime: Operation not permitted`
`tar: prokka-1.7/db/hmm: Cannot change mode to rwxr-xr-x: Operation not permitted`
 and many like this. I figured this was throw from this line

mr-c replied:
Did you try it? How does it halt? Anyone is allowed to chmod files that they own.

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -22,37 +22,56 @@ Make sure you edit your security groups to include port 22 (SSH) and port
+If running on EC2::
+ sudo  mkdir /mnt/bin
+ sudo chown ubuntu /mnt/bin
+ ln -s /mnt/bin ${HOME}/bin
```
I think these lines need more indention to be a literal quote; see https://github.com/ged-lab/2012-paper-diginorm/blob/non-root-install/pipeline/pipeline-notes.rst

mr-c commented on pipeline/install-prokka.sh:
```diff
@@ -6,41 +6,41 @@ cd /mnt
-./configure --prefix=/usr && make && make install
+./configure --prefix=${HOME} && make && make install
 
 cd /mnt
```
These lines could be the problem; depends on the permissions of the 'mnt' folder. Lets download source code and build inside `${HOME}/src`

mr-c commented on pipeline/install-prokka.sh:
```diff
@@ -6,41 +6,41 @@ cd /mnt
 tar xzf hmmer-3.1b1.tar.gz
 cd hmmer-3.1b1/
-./configure --prefix=/usr && make && make install
+./configure --prefix=${HOME} && make && make install
```
http://packages.ubuntu.com/search?keywords=hmmer :-)

mr-c commented on pipeline/install-prokka.sh:
```diff
@@ -6,41 +6,41 @@ cd /mnt
+cp prodigal ${HOME}/bin/
 
 cd /mnt
 curl -O ftp://ftp.ncbi.nih.gov/toolbox/ncbi_tools/converters/by_program/tbl2asn/linux64.tbl2asn.gz
```
http://packages.ubuntu.com/trusty/ncbi-tools-bin

themangoemoji committed:
Shake installs out of install-prokka script

* into apt-get lines

Conflicts:
	pipeline/pipeline-notes.rst

mr-c commented on pipeline/install-prokka.sh:
```diff
@@ -1,46 +1,10 @@
-cp aragorn /usr/local/bin
-
-cd /mnt
+cd ${HOME}/bin
```
source downloads should go into ${HOME}/src/${NAME}

themangoemoji replied:
@mr-c
should be cd ${HOME}/src/prodigal?

mr-c replied:
First line should be `set -e`

mr-c commented on pipeline/install-prokka.sh:
```diff
@@ -1,46 +1,10 @@
-./configure && make && make install
-
 
+cd ${HOME}/bin
```
source downloads should go into ${HOME}/src/${NAME}

themangoemoji replied:
@mr-c 
cd ${HOME}/src/prokka

mr-c commented on pipeline/install-prokka.sh:
```diff
@@ -1,46 +1,10 @@
-
-cd /mnt
+cd ${HOME}/bin
 curl -O http://prodigal.googlecode.com/files/prodigal.v2_60.tar.gz
```
Use https://github.com/hyattpd/Prodigal/archive/v2.6.1.tar.gz

themangoemoji committed:
Change install location to ~/src

themangoemoji committed:
Check out the correct branch, curl correct file

themangoemoji committed:
Check out the correct branch, curl correct file

themangoemoji committed:
Merge branch 'non-root-install' of https://github.com/ged-lab/2012-paper-diginorm into non-root-install

Conflicts:
	pipeline/install-prokka.sh

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -22,37 +22,61 @@ Make sure you edit your security groups to include port 22 (SSH) and port
+ cd ${HOME}/bin/
  curl -O http://www.ebi.ac.uk/~zerbino/velvet/velvet_1.2.10.tgz
  tar xzf velvet_1.2.10.tgz
  cd velvet_1.2.10
```
?

themangoemoji replied:
change to 
mkdir -p ${HOME}/src/velvet
cd  ${HOME}/src/velvet

themangoemoji committed:
Add rackspace instructions and prokka install updates

themangoemoji committed:
Merge branch 'non-root-install' of github.com:ged-lab/2012-paper-diginorm into non-root-install

themangoemoji commented:
after `make` in root@cloud-server-02:/mnt/data/2012-paper-diginorm#
LaTeX Warning: File `diginorm-coverage2-raw.pdf' not found on input line 1132.

mr-c commented on pipeline/install-prokka.sh:
```diff
@@ -1,46 +1,9 @@
-curl -O http://prodigal.googlecode.com/files/prodigal.v2_60.tar.gz
-tar xzf prodigal.v2_60.tar.gz
-cd prodigal.v2_60/
+cd ${HOME}/src/prodigal
```
Insert

```
#!/bin/bash
mkdir -p ${HOME}/src/prodigal
```

mr-c commented on pipeline/install-prokka.sh:
```diff
@@ -1,46 +1,9 @@
-./configure && make && make install
-
 
+cd ${HOME}/src/prokka
```
need to do a `mkdir -p` first here as well

mr-c commented:
- [ ] Put all root requiring commands together at the beginning
- [ ] only need to source virtualenv script once

themangoemoji committed:
Upgrade numpy

*Due ImportError on calc-r2.py

themangoemoji commented on pipeline/pipeline-notes.rst:
```diff
@@ -22,37 +22,71 @@ Make sure you edit your security groups to include port 22 (SSH) and port
+ source venv/bin/activate
  easy_install -U setuptools
  pip install khmer==1.1
+ pip install --upgrade numpy
```
@mr-c Added this line to fix:

`(venv)non-root@cloud-server-02:/mnt/data/2012-paper-diginorm/pipeline$ make`
`python calc-r2.py > calc-r2.txt`
`Traceback (most recent call last):`
`File "calc-r2.py", line 1, in <module>`
`import numpy, math, itertools`
`ImportError: No module named numpy`
`make: *** [calc-r2.txt] Error 1`

mr-c replied:
Is the python-numpy Debian package new enough to work?
http://packages.ubuntu.com/utopic/python-numpy

On Tue, Feb 10, 2015, 23:40 Michael Wright notifications@github.com wrote:

> In pipeline/pipeline-notes.rst
> https://github.com/ged-lab/2012-paper-diginorm/pull/6#discussion_r24453165
> :
> 
> >   easy_install -U setuptools
> >   pip install khmer==1.1
> > - pip install --upgrade numpy
> 
> Added this line due to:
> 
> (venv)non-root@cloud-server-02:/mnt/data/2012-paper-diginorm/pipeline$
> make
> python calc-r2.py > calc-r2.txt
> Traceback (most recent call last):
> File "calc-r2.py", line 1, in <module>
> import numpy, math, itertools
> ImportError: No module named numpy
> make: **\* [calc-r2.txt] Error 1
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ged-lab/2012-paper-diginorm/pull/6/files#r24453165.

mr-c replied:
Wrong URL, we're running Ubuntu Trusty:
http://packages.ubuntu.com/trusty/python-numpy

On Wed, Feb 11, 2015, 10:47 Michael Crusoe michael.crusoe@gmail.com wrote:

> Is the python-numpy Debian package new enough to work?
> http://packages.ubuntu.com/utopic/python-numpy
> 
> On Tue, Feb 10, 2015, 23:40 Michael Wright notifications@github.com
> wrote:
> 
> > In pipeline/pipeline-notes.rst
> > https://github.com/ged-lab/2012-paper-diginorm/pull/6#discussion_r24453165
> > :
> > 
> > >   easy_install -U setuptools
> > >   pip install khmer==1.1
> > > - pip install --upgrade numpy
> > 
> > Added this line due to:
> > 
> > (venv)non-root@cloud-server-02:/mnt/data/2012-paper-diginorm/pipeline$
> > make
> > python calc-r2.py > calc-r2.txt
> > Traceback (most recent call last):
> > File "calc-r2.py", line 1, in <module>
> > import numpy, math, itertools
> > ImportError: No module named numpy
> > make: **\* [calc-r2.txt] Error 1
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/ged-lab/2012-paper-diginorm/pull/6/files#r24453165.

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -62,20 +96,30 @@ Running the pipeline
 
+  
+ mkdir ~/src mkdir -p ~/src/prodigal
+ mkdir -p ~/src/prokka
```
117 + 118 aren't needed as they are in the install-prokka script

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -84,6 +128,9 @@ Once it successfully completes, copy the data over to the ../data/ directory::
+
+Now via root::
+
  ipython notebook --pylab=inline --no-browser --ip=* --port=80 &
```
s/80/8080/

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -84,6 +128,9 @@ Once it successfully completes, copy the data over to the ../data/ directory::
+
  ipython notebook --pylab=inline --no-browser --ip=* --port=80 &
 
 Connect into the ipython notebook (it will be running at 'http://<your EC2 hostname>'); if the above command succeeded but you can't connect in, you probably forgot to enable port 80 on your EC2 firewall.
```
`'http://<your EC2 hostname>:8080'`

themangoemoji committed:
Seperate EC2 and RackSpace instructions

*Change drive mounting and data storage to /home/user

themangoemoji committed:
Expand arguments for Makefile commands

*Issue #7

themangoemoji committed:
Add additional targets for missing files and download data

themangoemoji committed:
Rerout /mnt/bin to ${HOME}/bin

themangoemoji committed:
Rerout /mnt/bin to ${HOME}/bin

themangoemoji committed:
Resolve merge

themangoemoji committed:
Merge branch 'non-root-install' of https://github.com/ged-lab/2012-paper-diginorm into non-root-install

themangoemoji committed:
Update prokka-11 install

themangoemoji committed:
Merge with athyra

themangoemoji committed:
Merging athyra/local differences
Merge branch 'non-root-install' of https://github.com/ged-lab/2012-paper-diginorm into non-root-install

themangoemoji committed:
Update notes with proper root/non-root distinctions

themangoemoji committed:
Merge branch 'non-root-install' of github.com:ged-lab/2012-paper-diginorm into non-root-install

themangoemoji committed:
Fix install-prokka script

themangoemoji committed:
Fix install-prokka script

themangoemoji committed:
Merge branch 'non-root-install' of github.com:ged-lab/2012-paper-diginorm into non-root-install

Conflicts:
	pipeline/install-prokka.sh
	pipeline/pipeline-notes.rst

themangoemoji committed:
Merge branch 'non-root-install' of github.com:ged-lab/2012-paper-diginorm into non-root-install

Conflicts:
	pipeline/install-prokka.sh
	pipeline/pipeline-notes.rst

themangoemoji committed:
Merge branch 'non-root-install' of github.com:ged-lab/2012-paper-diginorm into non-root-install

themangoemoji committed:
Fixup pipeline notes

themangoemoji committed:
Move numpy install to root installs

themangoemoji committed:
Point root to install in user directory

themangoemoji committed:
Switch home to usert

themangoemoji commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,98 @@ and will then compile the paper from scratch using the new figures.
  easy_install -U setuptools
  pip install khmer==1.1
+ echo 'export PATH=${PATH}:${HOME}/bin' >> ${HOME}/.bashrc
+ source ${HOME}/.bashrc
```
@mr-c when I execute this source, the virtualenv is no longer activated. Is this a problem, so long as I repeat the step on line 93 (non root)? If I can grant non-root the right permissions I should be able to move the khmer out of root

mr-c replied:
The simplest thing to do would be to move lines 82-83 prior to lines 76

wrightmhw commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,98 @@ and will then compile the paper from scratch using the new figures.
 sizes)::
 
- cd /root
+ mkdir -p ${HOME}/src/velvet
```
I know we solved this before, but I'm having a hard time granting non-root the right permissions:

`miguel@cloud-server-02:~/src$ mkdir -p ${HOME}/src/velvet`
`mkdir: cannot create directory ‘/home/miguel/src/velvet’: Permission denied`

 and these also require permissions:

    echo 'export PATH=${PATH}:${HOME}/bin' >> ${HOME}/.bashrc
    source ${HOME}/.bashrc



an `ls -la` on home/non-root yeilds:

drwxr-xr-x 6 root root  4096 Mar 17 13:44 .
drwxr-xr-x 7 root root  4096 Mar 17 13:08 ..
drwxr-xr-x 2 root root  4096 Mar 17 13:43 bin
drwx------ 2 root root 16384 Mar 17 13:41 lost+found
drwxr-xr-x 2 root root  4096 Mar 17 13:44 src
drwxr-xr-x 6 root root  4096 Mar 17 13:43 venv

I ran `chmod -R u+rw /home/non-root`, shouldn't that cover this?

mr-c replied:
According to your transcript you are acting as user 'miguel' yet you are in `/home/non-root`. Why aren't you in `/home/miguel/` ?

themangoemoji commented on pipeline/Makefile:
```diff
@@ -350,27 +353,27 @@ ecoli_ref.fastq.bz2.keep.assembly.fa: ecoli_ref.fastq.bz2.keep.abundfilt.keep
 
 ecoli_dn.faa: ecoli_ref.fastq.bz2.keep.assembly.fa
-	/mnt/prokka-1.7/bin/prokka ecoli_ref.fastq.bz2.keep.assembly.fa --outdir ecoli_dn --prefix ecoli_dn --force
+	${HOME}/src/prokka/prokka-* ecoli_ref.fastq.bz2.keep.assembly.fa --outdir ecoli_dn --prefix ecoli_dn --force
```
@mr-c 
Here where I changed the prokka directories. I thought we talked about installing to `${HOME}/src/prokka/prokka-*/bin`, but both that and `${HOME}/src/prokka/prokka-*` throw the error below. Isn't this bin (next to the binaries) where we want these installs though? I don't think anywhere else is appropriate.

``` bash
/home/wrightm/src/prokka/prokka-* sar.keep.assembly.fa --outdir sar_dn --prefix sar_dn --force
/bin/sh: /home/wrightm/src/prokka/prokka-1.11: Is a directory
make: *** [sar_dn.faa] Error 126
```

mr-c replied:
How about using the path `${HOME}/src/prokka/prokka-*/bin/prokka` to match what was there before?

mr-c replied:
And according to https://github.com/ged-lab/2012-paper-diginorm/pull/6/files#diff-ba53c673353e6a7edb32d8c3bd0f6817R9 you shouldn't even have to use a full path, just call prokka

themangoemoji committed:
Add prodigal to root installs, change path variable ordering

themangoemoji committed:
Allow non-root to own their directory

themangoemoji committed:
Add screed install

wrightmhw commented on pipeline/install-prokka.sh:
```diff
@@ -1,46 +1,10 @@
-./configure && make && make install
-
-
+#!/bin/bash -ex
```
@mr-c If I got rid of prodigal, do I need this line?

mr-c replied:
Is this comment misplaced? You still should have the `#!` line for a shell script, yes.

themangoemoji commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,106 @@ and will then compile the paper from scratch using the new figures.
+Then we want to set the user directory to the home path, and grant permissions to the user::
+
+  HOME=/home/<your_user_name>
+  chown -R <your_user_name> ${HOME}
```
I had to use the recursive flag so that this would apply to all subdirectories.

mr-c replied:
Coo'.

themangoemoji commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,106 @@ and will then compile the paper from scratch using the new figures.
+
+Then we want to set the user directory to the home path, and grant permissions to the user::
+
+  HOME=/home/<your_user_name>
```
Before this line, root's `${HOME}` pointed to /user, and non-root's `${HOME}` pointed to /home/non-root, I found it way easier to make this change. Bad practice?

mr-c replied:
Um yeah, don't redefine the HOME environment variable. What are you trying to do here?

themangoemoji commented on pipeline/pipeline-notes.rst:
```diff
@@ -62,20 +121,17 @@ Running the pipeline
+ cd 2012-paper-diginorm/pipeline
  bash install-prokka.sh
- make /usr/local/share/khmer
+ make 
```
@mr-c I don't understand why this line is not working. Before I added lines 90 and 91, I had the problem `no module screed`. Are lines 90 and 91 incorrect ways to install screed? If not, I now have this error after make:

```
miguel@cloud-server-02:~/2012-paper-diginorm/pipeline$ make
load-into-counting.py -k 20 -N 4 -x 1e8 genome-reads.fa.ct genome-reads.fa
make: load-into-counting.py: Command not found
make: *** [genome-reads.fa.ct] Error 127
```

This makes me think this is a screed problem still.

mr-c replied:
See my above comment about root vs. non-root.

mr-c commented on pipeline/Makefile:
```diff
@@ -1,3 +1,6 @@
+#default to setting number of threads to number of processors
+THREADS != nproc
```
Did you test this? I think the syntax is

```
THREADS != $(shell nproc)
```

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,106 @@ and will then compile the paper from scratch using the new figures.
+
+If you need to create a new user, you can do so with::
+
+ adduser <your_user_name>
```
Inconsistent use of sudo in this section.

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,106 @@ and will then compile the paper from scratch using the new figures.
+
+ git clone git://github.com/ged-lab/screed.git
+ python setup.py install
+
```
Lines 75-91 belong in the 'non-root' section.

themangoemoji committed:
Move screed install to non-root

themangoemoji committed:
Correct thread variable syntax

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,105 @@ and will then compile the paper from scratch using the new figures.
-Next, we are going to set the instance up with many of the software 
-packages we will need::
+We will also want to link /mnt/bin directory to our home/bin directory::
+  sudo ln -s /mnt/bin ${HOME}/bin
```
When EC2 is available again, please test this out. Where does `/mnt/bin/` come from?

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,105 @@ and will then compile the paper from scratch using the new figures.
+If you are going to run this pipeline on a cloud machine, we recommend Amazon EC2 or RackSpace.
+These notes will give specific instructions to both of those machine types.
+
+EC2
```
I would title this 'Amazon EC2 setup'

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,105 @@ and will then compile the paper from scratch using the new figures.
+  sudo ln -s /mnt/bin ${HOME}/bin
+
+
+RackSpace
```
And likewise `RackSpace Setup`

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,105 @@ and will then compile the paper from scratch using the new figures.
+Formating/Mounting Drive::
+
+  mkfs -t ext4 /dev/xvde1
+  mount /dev/xvde1 /home/<your_user_name> -t ext4
```
idea: mount directly to `/home/`

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,105 @@ and will then compile the paper from scratch using the new figures.
+
+If you need to create a new user, you can do so with::
+
+  sudo adduser <your_user_name>
```
adduser after the mount below

mr-c replied:
- and then add them to the 'sudo' group: `sudo adduser <username> sudo`

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,105 @@ and will then compile the paper from scratch using the new figures.
+
+Then we want to set the user directory to the home path, and grant permissions to the user::
+
+  sudo chown -R <your_user_name> /home/<your_user_name>
```
given above two comments this won't be necessary

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,105 @@ and will then compile the paper from scratch using the new figures.
+
+  sudo chown -R <your_user_name> /home/<your_user_name>
 
 
```
finally, `su --login <your_user_name>`

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,105 @@ and will then compile the paper from scratch using the new figures.
+
+
+
+All Non-Root
```
'User land installs'

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,105 @@ and will then compile the paper from scratch using the new figures.
+All Non-Root
+--------------------------------------------
+Once you have completed all of the previous commands that require root permissions,
+go ahead and login as you normally would.
```
Drop this paragraph above

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,105 @@ and will then compile the paper from scratch using the new figures.
+Now we can go ahead and add ``~/bin`` to our path::
+
+ cd ${HOME}
+ source venv/bin/activate
```
huh? Drop this line

wrightmhw commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,105 @@ and will then compile the paper from scratch using the new figures.
 paper is currently using.::
- 
+
+ echo 'export PATH=${PATH}:${HOME}/bin' >> ${HOME}/.bashrc
```
these belong to line 79

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,105 @@ and will then compile the paper from scratch using the new figures.
- 
+
+ echo 'export PATH=${PATH}:${HOME}/bin' >> ${HOME}/.bashrc
+ source ${HOME}/.bashrc
```
lines 88-89 move up to line 82

mr-c replied:
sorry, make that lines 88-91

mr-c commented on pipeline/pipeline-notes.rst:
```diff
@@ -14,47 +14,105 @@ and will then compile the paper from scratch using the new figures.
+We also need to install screed::
+
+ git clone git://github.com/ged-lab/screed.git
+ python setup.py install
```
Drop this direct screed install

themangoemoji committed:
Use `prokka` directly in makefile, Update pipeline-notes

themangoemoji committed:
Fix user-land installs to use sudo
